snoop/walker.py

```diff
@@ -6,15 +6,16 @@
 
 class Walker(object):
 
-    def __init__(self, root, prefix, container_doc):
+    def __init__(self, root, prefix, container_doc, collection_id):
```

Comment:
I think it's ok to use a reference to a `Collection` instance instead of passing around its ID.